src/watch/mod.rs

```diff
@@ -102,17 +59,24 @@ impl WatchTunnel {
                 };
                 match resp {
                     Ok(Some(resp)) => {
-                        resp_sender.send(Ok(From::from(resp))).unwrap();
+                        resp_sender
+                            .send(Ok(From::from(resp)))
+                            .await
+                            .unwrap_or_else(|_| panic!());
                     }
                     Ok(None) => {
                         return;
                     }
                     Err(e) => {
-                        resp_sender.send(Err(From::from(e))).unwrap();
+                        resp_sender
+                            .send(Err(From::from(e)))
+                            .await
+                            .unwrap_or_else(|_| panic!());
                     }
                 };
             }
-        });
+        })
+        .detach();
```

Comment:
if detach, you might need to config how many pending tasks can run in parallel

Reply:
Yes, but for now, one etcd client will only have one WatchTunnel Task.